spec/moonshot/plugins/rotate_asg_instances/asg_spec.rb

```diff
@@ -262,4 +262,30 @@ def stub_cf_client
       end
     end
   end
+
+  describe '#instance_in_terminal_state?' do
+    context 'when an ec2 instance object is nil' do
+      let(:terminated_ec2_instance) do
+        Aws::EC2::Instance.new(id: "i-123456")
+      end
+      let(:check_terminal_state){ subject.send(:instance_in_terminal_state?, terminated_ec2_instance)}
```

Comment:
Let lazily sets variables. If someone adds a new test later it would not call the block again and would reuse the value from previous call. Here it looks like its being used like a proc.
Also not a best practice.

I recommend trying creating a new subject definition
```
subject { subject..send(:instance_in_terminal_state?, terminated_ec2_instance) }
if that does not work
subject { describes_class.new.send(:instance_in_terminal_state?, terminated_ec2_instance) }

Reply:
Updated